### PR TITLE
sdb_fmt() to setf_asm() in /arch/pic|/riscv|/z80

### PR DESCRIPTION
**Pull request by HN026**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

Partially addresses #1168 
...

### CODE REVIEW

File: librz/asm/arch/pic/pic_pic18.c [hunk 1]
```diff
@@ -119,9 +119,8 @@ int pic_pic18_disassemble(RzAsmOp *op, char *opbuf, const ut8 *b, int blen) {
 		return -1;
 	}
 	ut16 instr = rz_read_le16(b); // instruction
-	// if still redundan code is reported think of this of instr=0x2
-	const char *buf_asm = "invalid";
-	strcpy(opbuf, buf_asm);
+	// if still redundan code is reported think of this of instr=0x
```
Comment (XVilka) [inline librz/asm/arch/pic/pic_pic18.c]: `redundant`

File: librz/asm/arch/riscv/riscv.c [hunk 2]
```diff
@@ -36,23 +36,23 @@
...
 
 // TODO : an conf to chose between abi or numeric
-static const char * const *riscv_gpr_names = riscv_gpr_names_abi;
-static const char * const *riscv_fpr_names = riscv_fpr_names_abi;
+static const char *const *riscv_gpr_names = riscv_gpr_names_abi;
+static const char *const *riscv_fpr_names = riscv_fpr_names_abi;
 static int init = 0;
 
-static void arg_p (char *buf, unsigned long val, const char* const* array, size_t size) {
+static void arg_p(char *buf, unsigned long val, const char *const *array, size_t size) {
```
Comment (XVilka) [inline librz/asm/arch/riscv/riscv.c]: I wonder why it wasn't caught by the `clang-format`...

**Review by XVilka — COMMENTED** [review]
One minor nitpick, apart from that - LGTM.

File: librz/asm/arch/riscv/riscv.c (hunk 2 above)
Comment (HN026) [reply librz/asm/arch/riscv/riscv.c]: I was thinking the same, the file was oddly formatted

File: librz/asm/arch/pic/pic_baseline.c [hunk 3]
```diff
@@ -215,40 +215,41 @@ int pic_baseline_disassemble(RzAsmOp *op, char *opbuf, const ut8 *b, int l) {
...
 	case PIC_BASELINE_OP_ARGS_9K:
-		buf_asm = sdb_fmt("%s 0x%x", op_info->mnemonic, instr & PIC_BASELINE_OP_ARGS_9K_MASK_K);
+		rz_asm_op_setf_asm(op, "%s 0x%x", op_info->mnemonic, instr & PIC_BASELINE_OP_ARGS_9K_MASK_K);
+		break;
+	default:
+		rz_asm_op_set_asm(op, "invalid");
 		break;
 	}
-	strcpy(opbuf, buf_asm);
+	strcpy(opbuf, rz_asm_op_get_asm(op));
```
Comment (wargio) [inline librz/asm/arch/pic/pic_baseline.c]: refactor `asm_pic_disassemble` and remove `opbuf`

File: librz/asm/arch/riscv/riscv.c (hunk 2 above)
Comment (wargio) [reply librz/asm/arch/riscv/riscv.c]: > I wonder why it wasn't caught by the `clang-format`...

because it is gpl.

**Review by wargio — CHANGES_REQUESTED** [review]
please revert the formatting changes on `z80.c` and `riscv.c`
Those files are from GPL sources and should keep the same formatting.

**Comment by XVilka** [comment]
@wargio are you sure about `z80.c`? It seems LGPL one and uses `sdb_fmt` heavily. I think it should not be excluded from formatting checks

**Comment by wargio** [comment]
> @wargio are you sure about `z80.c`? It seems LGPL one and uses `sdb_fmt` heavily. I think it should not be excluded from formatting checks

From `sys/clang-format.py`

```py
skiplist = [
    "/gnu/",
    "librz/asm/arch/vax/",
    "librz/asm/arch/riscv/",
    "librz/asm/arch/sh/gnu/",
    "librz/asm/arch/i8080/",
    "librz/asm/arch/z80/",
```

**Comment by XVilka** [comment]
@HN026, please proceed with changes in `z80` and remove that directory from `sys/clang-format.py` skiplist.

**Review by wargio — APPROVED** [review]

**Comment by wargio** [comment]
@HN026 please re-run `sys/clang-format.py`

File: sys/clang-format.py [hunk 4]
```diff
@@ -29,7 +29,6 @@
     "librz/asm/arch/riscv/",
     "librz/asm/arch/sh/gnu/",
     "librz/asm/arch/i8080/",
-    "librz/asm/arch/z80/",
```
Comment (wargio) [inline sys/clang-format.py]: revert this. it is GPL3. only one file is LGPL3.

File: sys/clang-format.py
Comment (wargio) [inline sys/clang-format.py:1]: Seems that you have changed the permissions of this. can you set it back to `0755` ? just `chmod +x ./sys/clang-format.py`

File: librz/asm/arch/z80/z80.c [hunk 5]
```diff
@@ -68,64 +68,65 @@ static int z80OpLength (const ut8 *buf, int len) {
...
 	ut8 res;
 	if (!ret) {
 		return ret;
 	}
 	z_op = z80_op;
-	const char *buf_asm = "invalid";
 	switch (z_op[buf[0]].type) {
 	case Z80_OP8:
-		buf_asm = sdb_fmt ("%s", z_op[buf[0]].name);
+		rz_asm_op_setf_asm(op, "%s", z_op[buf[0]].name);
```
Comment (XVilka) [inline librz/asm/arch/z80/z80.c]: Just set those directly, here and in other similar places, no need to add unnecessary computations.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/asm/arch/z80/z80.c (hunk 5 above)
Comment (wargio) [inline librz/asm/arch/z80/z80.c]:
```suggestion
		rz_asm_op_set_asm(op, z_op[buf[0]].name);
```

File: librz/asm/arch/z80/z80.c [hunk 6]
```diff
@@ -68,64 +68,65 @@ static int z80OpLength (const ut8 *buf, int len) {
...
-	case Z80_OP8^Z80_ARG16:
-		buf_asm = sdb_fmt (z_op[buf[0]].name, buf[1]+(buf[2]<<8));
+	case Z80_OP8 ^ Z80_ARG16:
+		rz_asm_op_setf_asm(op, z_op[buf[0]].name, buf[1] + (buf[2] << 8));
 		break;
 	case Z80_OP16:
-		cb_tab = (const char **) z_op[buf[0]].op_moar;
-		buf_asm = sdb_fmt ("%s", cb_tab[buf[1]]);
+		cb_tab = (const char **)z_op[buf[0]].op_moar;
+		rz_asm_op_setf_asm(op, "%s", cb_tab[buf[1]]);
```
Comment (wargio) [inline librz/asm/arch/z80/z80.c]: same here

File: librz/asm/arch/z80/z80.c [hunk 7]
```diff
@@ -68,64 +68,65 @@ static int z80OpLength (const ut8 *buf, int len) {
...
 		break;
 	case Z80_OP_UNK ^ Z80_ENC1:
 		z_op = (const z80_opcode *)z_op[buf[0]].op_moar;
-		res = z80_ed_branch_index_res (buf[1]);
+		res = z80_ed_branch_index_res(buf[1]);
 		if (z_op[res].type == Z80_OP16) {
-			buf_asm = sdb_fmt ("%s", z_op[res].name);
+			rz_asm_op_setf_asm(op, "%s", z_op[res].name);
```
Comment (wargio) [inline librz/asm/arch/z80/z80.c]: same here

File: librz/asm/arch/z80/z80.c [hunk 8]
```diff
@@ -68,64 +68,65 @@ static int z80OpLength (const ut8 *buf, int len) {
...
+			rz_asm_op_setf_asm(op, z_op[res].name, buf[2] + (buf[3] << 8));
 		}
 		break;
 	case Z80_OP_UNK ^ Z80_ENC0:
 		z_op = (const z80_opcode *)z_op[buf[0]].op_moar;
-		res = z80_fddd_branch_index_res (buf[1]);
+		res = z80_fddd_branch_index_res(buf[1]);
 		if (z_op[res].type == Z80_OP16) {
-			buf_asm = sdb_fmt ("%s", z_op[res].name);
+			rz_asm_op_setf_asm(op, "%s", z_op[res].name);
```
Comment (wargio) [inline librz/asm/arch/z80/z80.c]: same

Comment (HN026) [reply librz/asm/arch/z80/z80.c]: Okay

**Review by XVilka — APPROVED** [review]

**Pull request merged**